### DISCUSSION TITLE
IDEA: Add the ability to sync disk volumes from LNMS for servers

### DISCUSSION THREAD

**Discussion by monteroman** [bonzo81/netbox-librenms-plugin#72] [Ideas]
I think that a great feature of this plugin would be to sync disk volumes of VMs and physical servers from LNMS to Netbox. This would help with updating Netbox with servers that have added new volumes or have updated disk sizes.

Thoughts?

**Comment by bonzo81** [comment]
Hi Monteroman,

Always happy to hear new feature ideas! This isn't one on my plan, mainly because I only Netbox for our network kit currently.

The key questions we would need to answer are:

1. What object in NetBox are we trying to model? What specific data do we need to create it?

2. Is that data available via the LibreNMS API?

Are you thinking specifically about the virtual disk object in Netbox? 

If you have a look at the [LibreNMS API](https://docs.librenms.org/API/) to see if what you are thinking about is available that would be a good start. 

If we know what we want to model, and the data exists then in theory this would be possible.

**Reply by monteroman** [reply]
I went through the API documentation and did some trial and error checking and there doesn't appear to be a way to pull that information from the API. The model would be for the virtual machines' virtual disks.

**Reply by bonzo81** [reply]
Thanks. Yeh I also had a look and couldn't see a way to get that info. If the data is in LibreNMS then it should be possible to make that available to the API, but that will need someone to submit a PR for it to the LIbreNMS project!
